ccm_web/client/src/components/UserEnrollmentForm.tsx

```diff
@@ -261,19 +257,7 @@ export default function UserEnrollmentForm (props: UserEnrollmentFormProps): JSX
 
   const renderForm = (): JSX.Element => {
     if (errorsWithContext.length > 0) {
-      const { error, context } = errorsWithContext[0]
-      const prefix = `An error occurred while ${context}`
-      let message
-      if (error instanceof CanvasError) {
-        message = (
-          <Typography>
-            {prefix}: {error.errors.length > 0 ? error.errors[0].message : 'unknown API error from Canvas.'}
-          </Typography>
-        )
-      } else {
-        message = <Typography key={0}>{prefix}: {error.message}.</Typography>
-      }
-      return <ErrorAlert messages={[message]} tryAgain={resetAll} />
+      return <ErrorAlert messages={[<APIErrorMessage key={0} {...errorsWithContext[0]} />]} tryAgain={resetAll} />
```

Comment:
Do you think, you need to revisit this when you integrate it checking for users exists or not for Non-UM feature?  there might be some change in Error response you were designing with this feature.

Reply:
Mmm... probably. That's the biggest question mark right now, is how do I handle the irregular data shape from errors. I think adding a type guard and creating a new error subclass (`ExternalUserError`?) would probably work, but I'll have to give it some thought. The internals of `APIErrorMessage` would probably need to be changed then, but at least that's just a change in one place.

Reply:
Ok, I won't vet on this now. If you want to revisit this when you are integrating the Non-UM user then I am fine with it.

Reply:
> I think adding a type guard and creating a new error subclass (ExternalUserError?)

This is good idea, since you know that the external user case is slightly different and easier for you to write code based on type of  error object.